From c6824719c131e0451723717f5dcf6f72862ac437 Mon Sep 17 00:00:00 2001
From: sagar sagar <sagar.sagar@oracle.com>
Date: Tue, 17 Sep 2024 13:06:20 +0000
Subject: [PATCH 3/5] persist directories: Pass using new bwrap --bind-fd
 option Instead of passing a /proc/self/fd bind mount we use --bind-fd, which
 has two advantages:  * bwrap closes the fd when used, so it doesn't leak into
 the started app  * bwrap ensures that what was mounted was the passed in fd
 (same dev/ino),    as there is a small (required) gap between symlink resolve
 and mount    where the target path could be replaced.

Please note that this change requires an updated version of bubblewrap.

Resolves: CVE-2024-42472
Co-authored-by: Simon McVittie <smcv@collabora.com>
Signed-off-by: Simon McVittie <smcv@collabora.com>

upstream ref:- 6bd603f6836e9b38b9b937d3b78f3fbf36e7ff75
Orabug: 37027734
Signed-off-by: sagar sagar <sagar.sagar@oracle.com>
---
 common/flatpak-context.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/flatpak-context.c b/common/flatpak-context.c
index 7f21139..8809bc9 100644
--- a/common/flatpak-context.c
+++ b/common/flatpak-context.c
@@ -2240,10 +2240,10 @@ flatpak_context_append_bwrap_filesystem (FlatpakContext  *context,
               continue;
             }
           
-          g_autofree char *src_via_proc = g_strdup_printf ("/proc/self/fd/%d", src_fd);
+          g_autofree char *src_via_proc = g_strdup_printf ("%d", src_fd);
          
           flatpak_bwrap_add_fd (bwrap, g_steal_fd (&src_fd));
-          flatpak_bwrap_add_bind_arg (bwrap, "--bind", src_via_proc, dest);
+          flatpak_bwrap_add_bind_arg (bwrap, "--bind-fd", src_via_proc, dest);
         }
     }
 
-- 
1.8.3.1

